File: BRUH/src/main/java/com/stelmo/brtest/MainActivity.java
package com.stelmo.brtest;

import android.app.Activity;
import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.LayerDrawable;
import android.graphics.drawable.ShapeDrawable;
import android.graphics.drawable.shapes.OvalShape;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;
import com.stelmo.wheelview.WheelView;
import com.stelmo.wheelview.adapter.WheelArrayAdapter;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Random;

public class MainActivity extends Activity {

    private static final int ITEM_COUNT = 6;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final WheelView wheelView = (WheelView) findViewById(R.id.wheelview);
        final Button fire_button = findViewById(R.id.fire_button);
        final Button reset_button = findViewById(R.id.reset_button);
        final int loss = losingItem();


        //create data for the adapter
        List<Map.Entry<String, Integer>> entries = new ArrayList<Map.Entry<String, Integer>>(ITEM_COUNT);
        for (int i = 0; i < ITEM_COUNT; i++) {
            Map.Entry<String, Integer> entry = MaterialColor.random(this, "\\D*_500$");
            entries.add(entry);
        }

        //populate the adapter, that knows how to draw each item (as you would do with a ListAdapter)
        wheelView.setAdapter(new MaterialColorAdapter(entries));

        //a listener for receiving a callback for when the item closest to the selection angle changes
        wheelView.setOnWheelItemSelectedListener(new WheelView.OnWheelItemSelectListener() {
            @Override
            public void onWheelItemSelected(WheelView parent, Drawable itemDrawable, int position) {
                //get the item at this position
                Map.Entry<String, Integer> selectedEntry = ((MaterialColorAdapter) parent.getAdapter()).getItem(position);
                parent.setSelectionColor(getContrastColor(selectedEntry));
            }
        });

        wheelView.setOnWheelItemClickListener(new WheelView.OnWheelItemClickListener() {
            @Override
            public void onWheelItemClick(WheelView parent, int position, boolean isSelected) {
                String msg;

                if(position == loss)
                    msg = "Loaded";

                else
                    msg = "Empty";

                Toast.makeText(MainActivity.this, msg, Toast.LENGTH_SHORT).show();
            }
        });

        fire_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                wheelView.stopWheel();
                if(wheelView.getSelectedPosition() == loss)
                    Toast.makeText(getApplicationContext(), "*BANG*", Toast.LENGTH_LONG).show();//display the text of button1
                else
                    Toast.makeText(getApplicationContext(), "*click*", Toast.LENGTH_LONG).show();//display the text of button1
            }
        });


        reset_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = getIntent();
                finish();
                startActivity(intent);
            }
        });

        //initialise the selection drawable with the first contrast color
        wheelView.setSelectionColor(getContrastColor(entries.get(0)));

    }

    // select a losing value TODO also select image from the losing player's gallery
    public int losingItem(){
        Random r = new Random();
        return r.nextInt(6);
    }

    //get the materials darker contrast
    private int getContrastColor(Map.Entry<String, Integer> entry) {
        String colorName = MaterialColor.getColorName(entry);
        return MaterialColor.getContrastColor(colorName);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == R.id.action_settings) {
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    static class MaterialColorAdapter extends WheelArrayAdapter<Map.Entry<String, Integer>> {
        MaterialColorAdapter(List<Map.Entry<String, Integer>> entries) {
            super(entries);
        }

        @Override
        public Drawable getDrawable(int position) {
            Drawable[] drawable = new Drawable[] {
                    createOvalDrawable(getItem(position).getValue()),
                    new TextDrawable(String.valueOf(position))
            };
            return new LayerDrawable(drawable);
        }

        private Drawable createOvalDrawable(int color) {
            ShapeDrawable shapeDrawable = new ShapeDrawable(new OvalShape());
            shapeDrawable.getPaint().setColor(color);
            return shapeDrawable;
        }
    }
}

File: BRUH/src/main/java/com/stelmo/brtest/MainActivity.kt
package com.stelmo.brtest

import android.Manifest
import android.app.Activity
import android.app.Dialog
import android.content.Intent
import android.content.pm.PackageManager
import android.graphics.Bitmap
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.graphics.drawable.Drawable
import android.graphics.drawable.LayerDrawable
import android.graphics.drawable.ShapeDrawable
import android.graphics.drawable.shapes.OvalShape
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.os.Environment
import android.os.Handler
import android.provider.MediaStore
import android.support.v4.app.ActivityCompat
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.*
import com.stelmo.wheelview.WheelView
import com.stelmo.wheelview.adapter.WheelArrayAdapter
import java.io.File

import java.util.ArrayList
import java.util.Random

class MainActivity : Activity() {

    //track player wins and losses
    var winCounter = 0
    var lossCounter = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //var randomImage = getRandomImage()

        var i = intent
        if(i.extras != null){
            winCounter += i.getIntExtra("wc", 0)
            lossCounter +=  i.getIntExtra("lc", 0)
        }

        var winView = findViewById(R.id.winStats) as TextView
        winView.setText("Wins: " + winCounter)

        var lossView = findViewById(R.id.lossStats) as TextView
        lossView.setText("Losses: " + lossCounter)

        System.out.println("Wins: " + winCounter)
        System.out.println("Losses: " + lossCounter)

        isReadStoragePermissionGranted()

        val wheelView = findViewById<View>(R.id.wheelview) as WheelView
        val fire_button = findViewById<Button>(R.id.fire_button)
        val reset_button = findViewById<Button>(R.id.reset_button)
        val menu_button = findViewById<Button>(R.id.menu_button)
        var wonSlots = mutableListOf<Int>()
        val loss = losingItem()


        //create data for the adapter, assign all slots same randomized color
        val entries = ArrayList<Map.Entry<String, Int>>(ITEM_COUNT)
        val entry = MaterialColor.random(this, "\\D*_500$")
        val empty_entry = MaterialColor.random(this, "\\D*_500$")

        println("color: " + entry)
        println("Empty color: " + empty_entry)

        for (i in 0 until ITEM_COUNT) {
            entries.add(entry)
        }

        //populate the adapter, that knows how to draw each item (as you would do with a ListAdapter)
        wheelView.adapter = MaterialColorAdapter(entries)

        //a listener for receiving a callback for when the item closest to the selection angle changes
        wheelView.setOnWheelItemSelectedListener { parent, itemDrawable, position ->
            //get the item at this position
            val selectedEntry = (parent.adapter as MaterialColorAdapter).getItem(position)
            parent.setSelectionColor(getContrastColor(selectedEntry))
        }


        //used to debug
        wheelView.onWheelItemClickListener = WheelView.OnWheelItemClickListener { parent, position, isSelected ->
//            val msg: String
//
//            if (position == loss)
//                msg = "Loaded"
//            else
//                msg = "Safe"
//
//            Toast.makeText(this@MainActivity, msg, Toast.LENGTH_SHORT).show()
        }

        fire_button.setOnClickListener {
            wheelView.stopWheel()
            if (wheelView.selectedPosition == loss){
                //Toast.makeText(applicationContext, "YOU LOSE!", Toast.LENGTH_SHORT).show()//display text on loss
                lossCounter++
                var intent = intent
                intent.putExtra("wc", winCounter)
                intent.putExtra("lc", lossCounter)
                finish()
                startActivity(intent)
                intent = Intent(baseContext, PopupActivity::class.java)
                startActivity(intent)
            }

            //if slot aleady played, mark as such with color and count
            else if(!wonSlots.contains(wheelView.selectedPosition)){
                //Toast.makeText(applicationContext, "You survived!", Toast.LENGTH_LONG).show()//display win text
                winCounter++
                wonSlots.add(wheelView.selectedPosition)
                entries[wheelView.selectedPosition] = empty_entry
                wheelView.adapter = MaterialColorAdapter(entries)

            }

            else
                Toast.makeText(applicationContext, "You have already played this slot", Toast.LENGTH_LONG).show()//no easy point farming

            System.out.println("Wins: " + winCounter)
            System.out.println("Losses: " + lossCounter)

            var winView = findViewById(R.id.winStats) as TextView
            winView.setText("Wins: " + winCounter)


            var lossView = findViewById(R.id.lossStats) as TextView
            lossView.setText("Losses: " + lossCounter)

        }


        reset_button.setOnClickListener {
            val intent = intent
            intent.putExtra("wc", winCounter)
            intent.putExtra("lc", lossCounter)
            finish()
            startActivity(intent)
        }

        menu_button.setOnClickListener {
            val intent = Intent(baseContext, HomeActivity::class.java)
            startActivity(intent)
        }

        //initialise the selection drawable with the first contrast color
        wheelView.setSelectionColor(getContrastColor(entries[0]))

    }

    // select a losing value
    fun losingItem(): Int {
        val r = Random()
        return r.nextInt(6)
    }

    //get the materials darker contrast
    private fun getContrastColor(entry: Map.Entry<String, Int>): Int {
        val colorName = MaterialColor.getColorName(entry)
        return MaterialColor.getContrastColor(colorName)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)
    }

    internal class MaterialColorAdapter(entries: ArrayList<Map.Entry<String, Int>>) : WheelArrayAdapter<Map.Entry<String, Int>>(entries) {

        override fun getDrawable(position: Int): Drawable {
            val drawable = arrayOf(createOvalDrawable(getItem(position).value), TextDrawable(position.toString()))
            return LayerDrawable(drawable)
        }

        private fun createOvalDrawable(color: Int): Drawable {
            val shapeDrawable = ShapeDrawable(OvalShape())
            shapeDrawable.paint.color = color
            return shapeDrawable
        }
    }

    //checks and requests device read perms
    fun isReadStoragePermissionGranted(): Boolean {
        if (Build.VERSION.SDK_INT >= 23) {
            if (checkSelfPermission(Manifest.permission.READ_EXTERNAL_STORAGE)
                    == PackageManager.PERMISSION_GRANTED) {
                return true
            } else {
                ActivityCompat.requestPermissions(this, arrayOf<String>(Manifest.permission.READ_EXTERNAL_STORAGE), 3)
                return false
            }
        }
        else { //permission is automatically granted on sdk<23 upon installation
            return true
        }
    }

    companion object {
        private val ITEM_COUNT = 6
    }
}



